### ISSUE TITLE
intellisense not work on WSL

### ISSUE THREAD

**Issue by laoshancun** [microsoft/vscode-cpptools#1830]
## problem
go to Declaration and peek Declaration not work.
eg.
function `ngx_redis_init` has beed defined in ngx_http_xcache_redis.h.
![tim 20180414110324](https://user-images.githubusercontent.com/4952748/38763689-7ed39674-3fd3-11e8-87fa-d3b75dc40e6a.jpg)
in file `ngx_http_xcache_module.c`,command `go to delaration` not work.
![tim 20180414110248](https://user-images.githubusercontent.com/4952748/38763690-7f1f9894-3fd3-11e8-84ca-4b7c92645d60.jpg)

## software
- OS:`windows10 enterprise 1803 (17133.73)`
- VSCODE : `1.22.2`
- C/C++:`0.16.1-preview`
## configuration
- c_cpp_properties.json
```json
{
    "configurations": [
        {
            "name": "WSL",
            "intelliSenseMode": "clang-x64",            
            "includePath": [
                "${workspaceRoot}",
                "${workspaceRoot}/src",
                "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/lib/gcc/x86_64-linux-gnu/6/include",
                "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/local/include",
                "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/local/nginx/include",
                "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/lib/gcc/x86_64-linux-gnu/6/include-fixed",
                "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/include/x86_64-linux-gnu",
                "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/include"
            ],
            "defines": [
                "__linux__",
                "__x86_64__"
            ],
            "browse": {
                "path": [
                    "${workspaceRoot}",
                    "${workspaceRoot}/src",
                    "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/lib/gcc/x86_64-linux-gnu/6/include",
                    "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/local/include",
                    "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/local/nginx/include",
                    "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/lib/gcc/x86_64-linux-gnu/6/include-fixed",
                    "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/include/x86_64-linux-gnu",
                    "${localappdata}/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/include"
                ],
                "limitSymbolsToIncludedHeaders": false,
                "databaseFilename": ""
            },
            "cStandard": "c11",
            "cppStandard": "c++17"
        }      
    ],
    "version": 3
}
```
## log
```text
IntelliSense Engine = Default.
The extension will use the Tag Parser for IntelliSense when #includes don't resolve.
Autocomplete is enabled.
Error squiggles are disabled.
File exclude: **/.git
File exclude: **/.svn
File exclude: **/.hg
File exclude: **/CVS
File exclude: **/.DS_Store
File exclude: **/.vscode
Search exclude: **/node_modules
Search exclude: **/bower_components
Search exclude: .git
Search exclude: .vscode
Search exclude: **/.vscode
Code browsing service initialized
  Unable to retrieve file system information for G:\gitlab\tengine-new\tengine. error = -1
  Unable to retrieve file system information for G:\gitlab\tengine-new\tengine/src. error = -1
  Unable to retrieve file system information for C:\Users\hualv\AppData\Local/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/lib/gcc/x86_64-linux-gnu/6/include. error = -1
  Unable to retrieve file system information for C:\Users\hualv\AppData\Local/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/local/include. error = -1
  Unable to retrieve file system information for C:\Users\hualv\AppData\Local/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/local/nginx/include. error = -1
  Unable to retrieve file system information for C:\Users\hualv\AppData\Local/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/lib/gcc/x86_64-linux-gnu/6/include-fixed. error = -1
  Unable to retrieve file system information for C:\Users\hualv\AppData\Local/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/include/x86_64-linux-gnu. error = -1
  Unable to retrieve file system information for C:\Users\hualv\AppData\Local/Packages/KaliLinux.54290C8133FEE_ey8k8hqnwqnmg/LocalState/rootfs/usr/include. error = -1
Discovering files...
  Discovering files: 0 file(s) processed
  0 file(s) removed from database
Done discovering files.
Parsing open files...
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
Done parsing open files.
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_module.c
Tag Parser will be used for IntelliSense operations in: file:///g%3A/GITLAB/TENGINE-NEW/TENGINE/DEPS/NGX_HTTP_XCACHE_MODULE/SRC/NGX_HTTP_XCACHE_MODULE.C
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_module.c
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_module.c
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_module.c
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_redis.c
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/ngx_http_file_cache_module.c
Shutting down IntelliSense server: G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\NGX_HTTP_FILE_CACHE_MODULE.C
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_module.c
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/ngx_http_file_cache_module.c
Shutting down IntelliSense server: G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\NGX_HTTP_FILE_CACHE_MODULE.C
Shutting down IntelliSense server: G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\SRC\NGX_HTTP_XCACHE_REDIS.C
Shutting down IntelliSense server: G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\SRC\NGX_HTTP_XCACHE_MODULE.C
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_redis.h
Failed to create IntelliSense client. Can't create intellisense client for G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\SRC\NGX_HTTP_XCACHE_REDIS.H
IntelliSense Engine = Default.
The extension will use the Tag Parser for IntelliSense when #includes don't resolve.
Autocomplete is enabled.
Error squiggles are enabled.
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_module.c
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ddebug.h
Failed to create IntelliSense client. Can't create intellisense client for G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\SRC\DDEBUG.H
Failed to create IntelliSense client. Can't create intellisense client for G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\SRC\DDEBUG.H
Failed to create IntelliSense client. Can't create intellisense client for G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\SRC\DDEBUG.H
Failed to create IntelliSense client. Can't create intellisense client for G:\GITLAB\TENGINE-NEW\TENGINE\DEPS\NGX_HTTP_XCACHE_MODULE\SRC\DDEBUG.H
Checking for syntax errors: file:///g%3A/gitlab/tengine-new/tengine/deps/ngx_http_xcache_module/src/ngx_http_xcache_redis.c
```

**Comment by sean-mcmanus** [comment]
Sounds like a duplicate of https://github.com/Microsoft/vscode-cpptools/issues/1694. WSL on the Windows Insiders build enabled case-sensitivity, which we don't handle correctly yet.

**Comment by sean-mcmanus** [comment]
This should be fixed with 0.17.0-insiders2 or later: https://github.com/Microsoft/vscode-cpptools/releases . The final release is planned for early next week.

**Comment by laoshancun** [comment]
great! :)
